design/threat-model/threat-model.md

```diff
@@ -0,0 +1,138 @@
+
+Kata Containers seeks to provide a second layer of isolation on top of those
+provided by traditional namespace-containers. The hardware virtualization interface is
+the basis of this additional layer. Kata launches a lightweight virtual machine, and
+uses the guest’s linux kernel to create a container workload, or workloads in the case of
+multi-container pods. In Kubernetes and in the Kata implementation, the sandbox is carried
+out at the pod level. In Kata, this sandbox is created using a virtual machine.
+
+todo: insert picture
+
+Kata is currently designed to protect host infrastructure from untrusted container workloads.
+From an attack point of view, we consider the actual container workload to be malicious.
+
+## Interface to Kata Containers: CRI, v2-shim, OCI
+
+A typical deployment of Kata Containers will be in Kubernetes by way of a CRI implementation.
+On every node, Kubelet will interact with a CRI implementor, which will in turn interface with
+an OCI based runtime, such as Kata Containers. Typical CRI implementors are cri-o and containerd.
+
+
+The CRI API, as defined at the Kubernetes [CRI-API repo](https://github.com/kubernetes/cri-api/),
+results in a few constructs being supported by the CRI implementation, and ultimately in the OCI
+runtime creating the workloads:
+
+
+
+Kata integrates with CRI implementations (CRIO and containerd) In order to run a container, several
+devices need to be exposed:
+
+These devices map to virtual machine devices and interfaces.
+
+Kata translates sandbox and container definitions to underlying virtualization technologies provided
+by a set of virtual machine monitors (VMMs) and hypervisors. These devices and their underlying
+implementations are discussed in detail in the following section.
+
+## Interface to the Kata sandbox/virtual machine
+
+In case of Kata, today the devices which we need in the guest are:
+virtio-blk and/or virtio-fs: In the current design of Kata Containers, we are reliant on the CRI implementor to assist in image handling and volume management on the host. As a result, we need to support a way of passing in rootfs, volumes, configmaps, etc, into the sandbox. Depending on how these are managed, a block based device  or file-system sharing is required. Kata Containers does this by way of virtio-blk/scsi or virtio-fs. In prior versions of Kata, virtio-9pfs was made available. This isn’t actively maintained upstream, is not performant, and is problematic from a POSIX compliance standpoint. As a result, we recommend users leverage virtio-blk/scsi and/or virtio-fs going forward
+Virtio-net: lalala, unless you use direct device assignment, blah blah
+Virtio-vsock: In order to interact with the guest agent and retrieve STDIO from containers, a medium of communication is required. This is available via virtio-vsock.
+
+Other interfaces which may be utilized include VFIO (direct device assignment) and ACPI for dynamic VM resource management (ie, hotplug).
```

Comment:
I havent yet looked at the PR for virtio-mem that has been opened recently, but just a reminder that needs to be mentioned and evaluated here as well.